### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by krastegar**
I was only able to connect spark and mariadb, but I was not able to run join query.

### CODE REVIEW

File: src/Week4_HW.py [hunk 1]
```diff
@@ -18,51 +20,105 @@ def set_conf():
     return conf
 
 
-def main():
+def read_data(table_name):
```
Comment (Zack1803) [inline src/Week4_HW.py:23]: Good strategy to use a function to load data...helps avoid redundancy

File: src/Week4_HW.py [hunk 2]
```diff
@@ -18,51 +20,105 @@ def set_conf():
...
+        """
+    )
+
+    tempTableDf.createOrReplaceTempView("temp")
+    tempTableDf.persist(StorageLevel.DISK_ONLY)
+
+    # Creating Rolling averag
+    rolling_avg_df = sqlContext.sql(
+        """
+    SELECT t.batter, CAST(t.local_date AS VARCHAR(30)),
```
Comment (Zack1803) [inline src/Week4_HW.py:96]: might be helful to seperate individual selections in the select statement...better readability

File: src/Week4_HW.py [hunk 3]
```diff
@@ -18,51 +20,105 @@ def set_conf():
...
+    )
+
+    # Putting Rolling_AVG into a transformer
+    # Have to make a column to feed into transformer
+    rolling_avg_df = rolling_avg_df.withColumn(
+        "arrayColumn", array("batter", "local_date", "rolling_avg")
+    )
+
+    # Count Vectorizer
+    count_vectorizer = CountVectorizer(
```
Comment (Zack1803) [inline src/Week4_HW.py:115]: I am not sure if count verctorizer is something that is expected from this assignment..

Comment (nmohieddin) [reply src/Week4_HW.py:115]: I think the count vectorizer would be for strings.

File: src/Week4_HW.py [hunk 4]
```diff
@@ -18,51 +20,105 @@ def set_conf():
...
+    sql_data = (
         sqlContext.read.format("jdbc")
         .option("url", mysql_jdbc_url)
         .option("driver", mysql_db_driver_class)
-        .option("dbtable", "batter_counts")
+        .option("dbtable", table_name)
         .option("user", user_name)
         .option("password", password)
         .load()
     )
```
Comment (nmohieddin) [inline src/Week4_HW.py:59]: Looks good!

File: src/Week4_HW.py [hunk 5]
```diff
@@ -18,51 +20,105 @@ def set_conf():
...
+
+    # Count Vectorizer
+    count_vectorizer = CountVectorizer(
+        inputCol="arrayColumn", outputCol="vectorized_values"
+    )
+    count_vectorizer_fitted = count_vectorizer.fit(rolling_avg_df)
+    rolling_avg_df = count_vectorizer_fitted.transform(rolling_avg_df)
+    rolling_avg_df.show()
 
-    jdbcDF.show()
```
Comment (nmohieddin) [inline src/Week4_HW.py]: Not sure what this is supposed to show

File: src/Week4_HW.py [hunk 6]
```diff
@@ -18,51 +20,105 @@ def set_conf():
...
     return
```
Comment (nmohieddin) [inline src/Week4_HW.py:123]: looks great!

**Pull request merged**